peer.go

```diff
@@ -289,28 +289,52 @@ func (p *Peer) HostPort() string {
 	return p.hostPort
 }
 
-// getActive returns a list of active connections.
-// TODO(prashant): Should we clear inactive connections?
-func (p *Peer) getActive() []*Connection {
-	var active []*Connection
-	p.runWithConnections(func(c *Connection) {
-		if c.IsActive() {
-			active = append(active, c)
+// getConn treats inbound and outbound connections as a single virtual list
+// that can be indexed. The peer must be read-locked.
+func (p *Peer) getConn(i int) *Connection {
+	inboundLen := len(p.inboundConnections)
+	if i < inboundLen {
+		return p.inboundConnections[i]
+	}
+
+	return p.outboundConnections[i-inboundLen]
+}
+
+func (p *Peer) getActiveConnLocked() (*Connection, bool) {
```

Comment:
One return is enough, but I think it's clearer that something can fail when it returns an ok bool as well as the return item. Otherwise, it's not clear whether the call can return nil or not.

this makes it harder to ignore the `nil` case since you can't do something like `getActiveConn().Close()`